fdbclient/NativeAPI.actor.h

```diff
@@ -305,10 +285,10 @@ class Transaction : NonCopyable {
 
 	double backoff;
 	Version committedVersion;
-	CommitTransactionRequest tr;
+	CommitTransactionRequest* tr;
```

Comment:
We could consider making this a `Reference` type so that we don't need to explicitly delete it. I notice this type is used in a lot of places and I don't know if any others could benefit from holding them by reference, which might make this change more compelling.

Reply:
I don't think this is ever shared. `Reference<CommitTransactionRequest>` is awkward to make a forward declaration for. What we probably want is `std::unique_ptr` but that also can't be forward declared (as the destructor needs to know the size). I think using a pointer here is the easiest, but I can also use some template hacks to use `Reference` here to make this a bit safer if you prefer that.